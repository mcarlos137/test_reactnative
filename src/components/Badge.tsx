import React from 'react';
import { StyleSheet, Text, Image } from 'react-native';

const Badge = ({ text }: { text: string }) => {

  return (
    <>
      <Image
        source={require('../assets/images/user.png')}
      />
      <Text style={styles.text} >{text}</Text>
    </>
  );
}

export default React.memo(Badge)

const styles = StyleSheet.create({
  text: {
    fontSize: 24,
    marginTop: 20,
    marginBottom: 20,
    fontWeight: 500,
    fontFamily: 'Poppins'
  },
});
