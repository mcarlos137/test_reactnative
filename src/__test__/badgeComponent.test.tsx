import * as React from 'react';
import renderer from 'react-test-renderer';
//COMPONENTS
import Badge from '../components/Badge';

it(`renders correctly`, () => {
  const tree = renderer.create(<Badge text='Hi, Carlos' />).toJSON();
  expect(tree).toMatchSnapshot();
});
