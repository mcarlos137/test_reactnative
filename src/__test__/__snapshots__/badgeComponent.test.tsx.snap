// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/user.png",
      }
    }
  />,
  <Text
    style={
      {
        "fontFamily": "Poppins",
        "fontSize": 24,
        "fontWeight": 500,
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    Hi, Carlos
  </Text>,
]
`;
